overlord/servicestate/quota_handlers.go

```diff
@@ -620,3 +620,45 @@ func validateSnapForAddingToGroup(st *state.State, snaps []string, group string,
 
 	return nil
 }
+
+func quotaControlAffectedSnaps(t *state.Task) (snaps []string, err error) {
+	qcs := []QuotaControlAction{}
+	if err := t.Get("quota-control-actions", &qcs); err != nil {
+		return nil, fmt.Errorf("internal error: cannot get quota-control-action: %v", err)
+	}
+
+	// if state-updated was already set we can use it
+	var updated quotaStateUpdated
+	if err := t.Get("state-updated", &updated); err != state.ErrNoState {
+		if err != nil {
+			return nil, err
+		}
+		// TODO: consider boot-id as well?
```

Comment:
IMHO I think checking the boot-id makes sense, if the task in state has been updated to mention the boot-id, then it is almost done and if there is an old boot-id there, then it means that we rebooted and so there is exactly nothing left for the task to do and as such, nothing to conflict with a new change affecting a snap in the group that was part of this change

Reply:
I'll consider this for a follow-up